.github/workflows/build.staging.yml

```diff
@@ -0,0 +1,73 @@
+name: Build & Deploy Staging CI
+
+on:
+  push:
+    branches: [staging]
+
+jobs:
+  build_bp_gallery:
+    name: Build & Deploy bp-gallery
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./projects/bp-gallery
+
+    strategy:
+      matrix:
+        node-version: [16.x]
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Setup Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+      - name: Generate APIConnector
+        run: yarn generate-api
+      - name: Build bp-gallery
+        run: yarn build
+      - name: Deploy build files on the server
+        uses: appleboy/scp-action@master
+        with:
+          host: ${{ secrets.REMOTE_HOST }}
+          username: ${{ secrets.REMOTE_USER }}
+          key: ${{ secrets.SSH_KEY }}
+          source: "projects/bp-gallery/build/"
+          strip_components: 3
+          target: ${{ secrets.REMOTE_STAGING_TARGET }}
+          rm: true
+  build_bp_strapi:
+    name: Build & Deploy bp-strapi
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./projects/bp-strapi
+
+    strategy:
+      matrix:
+        node-version: [16.x]
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Setup Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Set up SSH
+        run: |
+          mkdir -p ~/.ssh/
+          echo "$SSH_KEY" > ./deploy.key
+          sudo chmod 600 ./deploy.key
+          echo "$SSH_KNOWN_HOSTS" > ~/.ssh/known_hosts
+        shell: bash
+        env:
+          SSH_KEY: ${{secrets.SSH_KEY}}
+          SSH_KNOWN_HOSTS: ${{secrets.SSH_KNOWN_HOSTS}}
+      - name: Install PM2 for deployment
+        run: yarn add pm2
```

Comment:
We already have a `pm2` dependency in the package.json, maybe a `yarn install` would be more explicit or global add?